internal/internal_workflow_client_test.go

```diff
@@ -1443,3 +1703,150 @@ func (m *PartialCancelRequestMatcher) Matches(a interface{}) bool {
+				ExecutionStartToCloseTimeout:    10 * time.Second,
+				DecisionTaskStartToCloseTimeout: 5 * time.Second,
+				DelayStart:                      0 * time.Second,
+				JitterStart:                     0 * time.Second,
+			},
+			workflowFunc: func(ctx Context) {},
+			wantErr:      "missing TaskList",
+		},
+		{
+			name: "invalid ExecutionStartToCloseTimeout",
+			options: StartWorkflowOptions{
+				ID:                              workflowID,
+				TaskList:                        tasklist,
+				ExecutionStartToCloseTimeout:    0 * time.Second, // this causes error
+				DecisionTaskStartToCloseTimeout: 5 * time.Second,
+				DelayStart:                      0 * time.Second,
+				JitterStart:                     0 * time.Second,
+			},
+			workflowFunc: func(ctx Context) {},
+			wantErr:      "missing or invalid ExecutionStartToCloseTimeout",
+		},
+		{
+			name: "negative DecisionTaskStartToCloseTimeout",
+			options: StartWorkflowOptions{
+				ID:                              workflowID,
+				TaskList:                        tasklist,
+				ExecutionStartToCloseTimeout:    10 * time.Second,
+				DecisionTaskStartToCloseTimeout: -1 * time.Second, // this causes error
+				DelayStart:                      0 * time.Second,
+				JitterStart:                     0 * time.Second,
+			},
+			workflowFunc: func(ctx Context) {},
+			wantErr:      "negative DecisionTaskStartToCloseTimeout provided",
+		},
+		{
+			name: "negative DelayStart",
+			options: StartWorkflowOptions{
+				ID:                              workflowID,
+				TaskList:                        tasklist,
+				ExecutionStartToCloseTimeout:    10 * time.Second,
+				DecisionTaskStartToCloseTimeout: 5 * time.Second,
+				DelayStart:                      -1 * time.Second, // this causes error
+				JitterStart:                     0 * time.Second,
+			},
+			workflowFunc: func(ctx Context) {},
+			wantErr:      "Invalid DelayStart option",
+		},
+		{
+			name: "negative JitterStart",
+			options: StartWorkflowOptions{
+				ID:                              workflowID,
+				TaskList:                        tasklist,
+				ExecutionStartToCloseTimeout:    10 * time.Second,
+				DecisionTaskStartToCloseTimeout: 5 * time.Second,
+				DelayStart:                      0 * time.Second,
+				JitterStart:                     -1 * time.Second, // this causes error
+			},
+			workflowFunc: func(ctx Context) {},
+			wantErr:      "Invalid JitterStart option",
+		},
+		{
+			name: "invalid workflow func",
+			options: StartWorkflowOptions{
+				ID:                              workflowID,
+				TaskList:                        tasklist,
+				ExecutionStartToCloseTimeout:    10 * time.Second,
+				DecisionTaskStartToCloseTimeout: 5 * time.Second,
+				DelayStart:                      0 * time.Second,
+				JitterStart:                     0 * time.Second,
+			},
+			workflowFunc: func(ctx Context, a, b int) {}, // this causes error because args not provided
+			wantErr:      "expected 2 args for function",
```

Comment:
tbh I think I'd pass an explicitly-empty args, or just one arg or something.

relying on the zero value when we haven't seen it for over 100 lines (it's in fact completely unused in ALL of these tests) seems a bit strange.

Reply:
yep agreed. I missed putting explicitly empty args